### ISSUE TITLE
Support min/max callee count on callables.

### ISSUE THREAD

**Issue by tmontes** [tmontes/python-wires#3]
Would it be useful somehow limiting the min/max number of wired callees?

Maybe it could help ensure some "conditions" are always met. Examples:
* Wiring can raise an exception if the maximum wired callees is reached; (use case: ensure no more than one wiring is in place, preventing coding mistakes, for example)
* Unwiring can raise an exception if the minimum wired callees is reached.
* Calling a wired callable with less callees than the minimum callees can also raise an exception.

Food for thought.

If moving forward:
* A simple and flexible API must be created that works with both `Wiring` instances and the `wiring` singleton.
* Should it be set at the `Wiring` level? Or at the `callable` level? Both?

**Comment by tmontes** [comment]
API possibility:

**Wiring level min/max callee setting**

(only applicable to Wiring instances, not to the global Singleton, which has no default callee limits)


```
w = Wiring(min_callees=<min>, max_callees=<max>)    # both defaulting to None, meaning no limit
```

**Setting wiring per-callable min/max callees**

```
w = Wiring()

w.this.min_callees = 1
w.this.max_callees = 1

w.this()    # Raises an exception: less than min_callees wired

w.wire.this.calls_to(<first-callee>)
w.this()

w.wire.this.calls_to(<second-callee>)    # raises an exception: max_callees reached

w.unwire.this.calls_to(<first-callable>)    # raises an exception: min_callees reached
````

**Comment by tmontes** [comment]
More:
* Maybe there can be a `wiring_count` per callable read-only attribute, returning the number of wired callees for a given callable? It could even be accessible via `__len__`?
* And/or, there can be a `wirings` per callable read-only attribute returning a list of (\<callee>, \<callee-args>, \<callee-kwargs>)? Or maybe callables can be iterable, producing such tuples?

**Comment by tmontes** [comment]
In the comment above I stated that this code would raise an exception given that less than `min_callees` were wired. How should this behave depending on caller/callee coupling mode?
```
w = Wiring()
w.this.min_callees = 1
w.this()
```

Possibilities:
* `min_callee` setting has precedence: this case raises an exception regardless of coupling.
* `coupling` has precedence: does not raise exception when decoupled, raises when coupled.

Thoughts: `min_callee` vs `coupling` work at different levels:
* The first operates at the wiring level (do we have callees? yes, no, how many?)
* The second operates *when* there are callees, at call-time, coupling or decoupling caller and callees.

Given this, I'm inclined to say that `min_callees` should, in fact, have precedence and raise an exception regardless of `coupling`; with #7 this will need further thinking (but we'll leave it at that, for now).

**Comment by tmontes** [comment]
WIP in #27.

**Comment by tmontes** [comment]
Mostly implemented in #27.
Pending ideas transferred to #28.
